cfgov/unprocessed/apps/youth-employment-success/js/util.js

```diff
@@ -0,0 +1,56 @@
+let UNDEFINED;
+
+const REDUCER_RETURN_ERROR = 'Reducer must return a state object';
+const processStateFromReducers = reducers => (
+  state = UNDEFINED,
+  action = { type: null }
+) => {
+  /**
+   * If reducers is not an object
+   */
+  const reducersKeys = Object.keys( reducers );
+  const finalState = {};
+  let hasChanged = false;
+
+  for ( let i = 0; i < reducersKeys.length; i++ ) {
+    const key = reducersKeys[i];
+    const reducer = reducers[key];
+
+    /**
+     * if we didnt already get a composite state object, e.g. if this is
+     * the first time the reducers are running, default to undefined
+     * and force the reducer to use it's default state
+     */
+    const previousState = state ? state[key] : state;
+    let argumentsToApply = [ action ];
+    argumentsToApply = [ previousState, ...argumentsToApply ];
```

Comment:
Could this line be combined into:
```
let argumentsToApply = [ previousState, action ];
```
Or am I missing something on what the spread is doing here?

Reply:
Nope! I wrote this pretty quickly, that can be changed for sure.